src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -122,17 +123,34 @@ public override bool TryGetVariable<T>(string name, out T value)
 
             if (variable != null)
             {
+                object outVal;
```

Comment:
nit: ideally the duplicate code should be moved to a utility method. Also, the switch statement can be avoid to simply the code.
```c#
            if (variable != null)
            {
                object outVal = variable.Value;
                if (outVal is PSObject psobject)
                {
                    if (psobject.BaseObject is PSCustomObject)
                    {
                        Dictionary<string, object> table = new Dictionary<string, object>();
                        foreach (var p in psobject.Properties)
                        {
                            table.Add(p.Name, p.Value);
                        }
                        outVal = table;
                    }
                    else
                    {
                        outVal = psobject.BaseObject;
                    }
                }

                if(outVal is T tObj)
                {
                    value = tObj;
                    return true;
                }
            }
```